### ISSUE TITLE
Selected ListItem styles not working with SSR

### ISSUE THREAD

**Issue by dan-kwiat** [material-components/material-components-web-react#492]
With client-side rendering the first item gets the `mdc-list-item--selected` class and `aria-selected="true"` as expected.  However with server-side rendering (tested with next.js) this is not the case (and all the items look the same).  I believe it's an issue with this library rather than material-components-web?

```jsx
<List
  singleSelection
  selectedIndex={0}
>
  <ListItem>
    <ListItemText primaryText='One' />
  </ListItem>
  <ListItem>
    <ListItemText primaryText='Two' />
  </ListItem>
  <ListItem>
    <ListItemText primaryText='Three' />
  </ListItem>
</List>
```

**Comment by moog16** [comment]
@dan-kwiat yes from my understanding of how SSR works, I believe the problem is with MDC React and not Web. I'm not super familiar with SSR. 

Do you have a test project that I can look at/learn from with the List HTML shown above?

**Comment by dan-kwiat** [comment]
@moog16 to be honest I'm pretty new to SSR as well.  This example has a List in the Drawer (see DrawerList.js): https://github.com/dan-kwiat/material-next-template

In development the first item is selected as expected (see screenshot below) and it responds to item clicks.  In next's static build however no item is selected (see demo) and `handleSelect` is never triggered.

![screenshot 2018-12-07 at 09 34 57](https://user-images.githubusercontent.com/5648607/49639648-6cae2c80-fa03-11e8-89ae-3b27d09d8ff1.png)

**Comment by moog16** [comment]
I visited your demo https://dan-kwiat.github.io/material-next-template/ and downloaded your project. I noticed that the `setSelected` method is not being called either, which leads me to believe that your environment isn't properly setup. 

In your dev setup, clicking on a list item console.log(), but this does not happen in your live demo.

**Comment by dan-kwiat** [comment]
@moog16 it isn't being called because `List`'s `handleSelect` method isn't triggered.  Setting the state in an explicit click event handler on `ListItem` works in both cases (see updated demo & code) but the active class still isn't applied.

**Comment by dan-kwiat** [comment]
@moog16 `componentDidMount` is not executed server side but still gets called on first rendering by the client.

I think the problem is the use of `child.type.name` in @material/react-list.  The type name for `<ListItem>` children is `"ListItem"` with CSR but `"t"` for SSR (see updated demo logs).  Therefore with SSR `this.listItemCount` is always zero and so is `index` in all the event handlers.

This explains why the logic to apply the selected class in the foundation's `setSelectedIndex` method isn't executed.  Is there a better way to get list item indexes - maybe like `getListItemIndex_` in @material/list?

Thanks for your time on this.

**Comment by dan-kwiat** [comment]
@moog16 so this is a code minification problem, nothing to do with SSR.  Same problem with a create-react-app build.

A workaround would be to define the list item type at the top of [index.js](https://github.com/material-components/material-components-web-react/blob/master/packages/list/index.js) as:
```jsx
const ListItemType = (<ListItem/>).type
```
then replace the condition in `renderChild`:
```jsx
if (child.type.name === 'ListItem') {...} // current condition fails with code minification
if (child.type === ListItemType) {...} // this should work
```

More on element type comparisons here: https://github.com/gaearon/react-hot-loader/issues/304

Should I submit a PR with the above?

**Comment by dan-kwiat** [comment]
#509

**Comment by moog16** [comment]
@dan-kwiat thanks for figuring that out! I spent quite a bit of time and couldn't figure it out. I'll try testing this out with your SSR demo.

**Comment by moog16** [comment]
Merged #509...thanks again!
